ubuntu_image/tests/test_main.py

```diff
@@ -398,6 +410,101 @@ def test_does_not_fit(self):
             call('Volume contents do not fit (72B over): '
                  'volumes:<pc>:structure:<mbr> [#0]'))
 
+    def test_hook_fired(self):
+        # For the purpose of testing, we will be using the post-populate-rootfs
+        # hook as we made sure it's still executed as part of of the
+        # DoNothingBuilder.
+        hookdir = self._resources.enter_context(TemporaryDirectory())
+        hookfile = os.path.join(hookdir, 'post-populate-rootfs')
+        # Let's make sure that, with the use of post-populate-rootfs, we can
+        # modify the rootfs contents.
+        with open(hookfile, 'w') as fp:
+            fp.write("""\
+#!/bin/sh
+touch $UBUNTU_IMAGE_HOOK_ROOTFS/foo
+""")
+        os.chmod(hookfile, 0o744)
+        workdir = self._resources.enter_context(TemporaryDirectory())
+        self._resources.enter_context(patch(
+            'ubuntu_image.__main__.ModelAssertionBuilder',
+            DoNothingBuilder))
+        code = main(('--hooks-directory', hookdir,
+                     '--workdir', workdir,
+                     self.model_assertion))
+        self.assertEqual(code, 0)
+        self.assertTrue(os.path.exists(os.path.join(workdir, 'root', 'foo')))
+        # XXX: Best if we could also easily inspect the resulting image file if
+        # the file is there.
```

Comment:
How about being barbarians here and do the following:

- put a unique string in the $UBUNTU_IMAGE_HOOK_ROOTFS/foo file
- grep or otherwise binary scan the resulting image file for that unique string

Subtlety bad. Axe good! Waaaaaag!